chanzuckerberg/czgenepi
# Update Modal

### Description

1. Extract `AgreeTerm` page's Dialog into a common component that will be moved into `czifui` after V2
2. Add Dialog support for `narrow` prop, since we need it for the Tree
2. Use the Dialog for both modals in our app, outlink to Nextstrain.org and `AgreeTerm` page

#### Issue
https://app.clubhouse.io/genepi/story/136483/convert-the-tree-linking-frontend-component-box-into-a-real-modal

### Test plan

1. Visit https://thuang-update-modal-frontend.dev.genepi.czi.technology/data/phylogenetic_trees
2. Log in as one of the test accounts
3. Clicking on the Tree Name cell in any row should pop up the dialog
4. If you scroll all the way to the bottom of the table and click on another Tree Name, the modal should still work
5. In RDEV, it seems like out link to Nextstrain.org doesn't work?

<img width="1684" alt="Screen Shot 2021-05-12 at 6 04 11 PM" src="https://user-images.githubusercontent.com/6309723/118064513-00605180-b350-11eb-9b1c-e100cadc482f.png">

![demo](https://user-images.githubusercontent.com/6309723/118064568-1f5ee380-b350-11eb-9196-d4d4e5d1263b.gif)
